.github/workflows/prow-pr-automerge.yml

```diff
@@ -0,0 +1,17 @@
+# This Github workflow will check every hour for PRs with the lgtm label and will attempt to automatically merge them.
+# If the hold label is present, it will block automatic merging.
+
+name: "Prow merge on lgtm label"
+on:
+  schedule:
+  - cron: "0 * * * *" # every hour
```

Comment:
Is there a way to trigger it after a comment or when the label is added? Like k8s prow?

Reply:
We can trigger the workflow when `lgtm` label is added to the PR by doing something like below.
```yaml
name: "Prow merge on lgtm label"
on:
  pull_request:
    types:
      - labeled

jobs:
  auto-merge:
    runs-on: ubuntu-latest
    if: contains(github.event.label.name, 'lgtm')
    steps:
      - uses: jpmcb/prow-github-actions@v1.1.3
        with:
          jobs: 'lgtm'
          github-token: "${{ secrets.GITHUB_TOKEN }}"
          merge-method: 'squash'
```
But there are limitations (that i know of).
1. The 'lgtm' label is added by the `github-actions` bot when we comment `/lgtm`. This won't trigger the workflow. This is a limitation of github. We can overcome this by using a `PAT Token` instead of `GITHUB_TOKEN`. You can read the detailed explanation here https://github.com/orgs/community/discussions/25565

2. Let's take a scenario where a PR is given `/lgtm` but not approved. Since `lgtm` label is added, the workflow will be triggered but the merge will fail as the PR is not approved. After you approved the PR, you have to remove the lgtm label and re add it to trigger the workflow.